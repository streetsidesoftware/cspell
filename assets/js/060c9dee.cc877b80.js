"use strict";(self.webpackChunkwebsite=self.webpackChunkwebsite||[]).push([[94023],{46039:(e,t,n)=>{n.d(t,{R:()=>l,x:()=>c});var i=n(18225);const s={},r=i.createContext(s);function l(e){const t=i.useContext(r);return i.useMemo((function(){return"function"==typeof e?e(t):{...t,...e}}),[t,e])}function c(e){let t;return t=e.disableParentContext?"function"==typeof e.components?e.components(s):e.components||s:l(e.components),i.createElement(r.Provider,{value:t},e.children)}},87257:(e,t,n)=>{n.r(t),n.d(t,{assets:()=>a,contentTitle:()=>c,default:()=>p,frontMatter:()=>l,metadata:()=>i,toc:()=>o});const i=JSON.parse('{"id":"api/cspell-lib/functions/readRawSettings","title":"Function: readRawSettings()","description":"readRawSettings(filename, relativeTo?): Promise\\\\","source":"@site/docs/api/cspell-lib/functions/readRawSettings.md","sourceDirName":"api/cspell-lib/functions","slug":"/api/cspell-lib/functions/readRawSettings","permalink":"/docs/api/cspell-lib/functions/readRawSettings","draft":false,"unlisted":false,"editUrl":"https://github.com/streetsidesoftware/cspell/tree/main/website/docs/docs/api/cspell-lib/functions/readRawSettings.md","tags":[],"version":"current","frontMatter":{},"sidebar":"tutorialSidebar","previous":{"title":"Function: readFileSync()","permalink":"/docs/api/cspell-lib/functions/readFileSync"},"next":{"title":"Function: readSettings()","permalink":"/docs/api/cspell-lib/functions/readSettings"}}');var s=n(37557),r=n(46039);const l={},c="Function: readRawSettings()",a={},o=[{value:"Parameters",id:"parameters",level:2},{value:"filename",id:"filename",level:3},{value:"relativeTo?",id:"relativeto",level:3},{value:"Returns",id:"returns",level:2}];function d(e){const t={a:"a",blockquote:"blockquote",code:"code",h1:"h1",h2:"h2",h3:"h3",header:"header",p:"p",strong:"strong",...(0,r.R)(),...e.components};return(0,s.jsxs)(s.Fragment,{children:[(0,s.jsx)(t.header,{children:(0,s.jsx)(t.h1,{id:"function-readrawsettings",children:"Function: readRawSettings()"})}),"\n",(0,s.jsxs)(t.blockquote,{children:["\n",(0,s.jsxs)(t.p,{children:[(0,s.jsx)(t.strong,{children:"readRawSettings"}),"(",(0,s.jsx)(t.code,{children:"filename"}),", ",(0,s.jsx)(t.code,{children:"relativeTo"}),"?): ",(0,s.jsx)(t.code,{children:"Promise"}),"<",(0,s.jsx)(t.a,{href:"/docs/api/cspell-lib/interfaces/CSpellSettingsWithSourceTrace",children:(0,s.jsx)(t.code,{children:"CSpellSettingsWithSourceTrace"})}),">"]}),"\n"]}),"\n",(0,s.jsxs)(t.p,{children:["Defined in: ",(0,s.jsx)(t.a,{href:"https://github.com/streetsidesoftware/cspell/blob/3fc3ef2ea9def847ef20287c6238ff423e6493eb/packages/cspell-lib/src/lib/Settings/Controller/configLoader/defaultConfigLoader.ts#L99",children:"packages/cspell-lib/src/lib/Settings/Controller/configLoader/defaultConfigLoader.ts:99"})]}),"\n",(0,s.jsx)(t.h2,{id:"parameters",children:"Parameters"}),"\n",(0,s.jsx)(t.h3,{id:"filename",children:"filename"}),"\n",(0,s.jsxs)(t.p,{children:[(0,s.jsx)(t.code,{children:"string"})," | ",(0,s.jsx)(t.code,{children:"URL"})]}),"\n",(0,s.jsx)(t.h3,{id:"relativeto",children:"relativeTo?"}),"\n",(0,s.jsxs)(t.p,{children:[(0,s.jsx)(t.code,{children:"string"})," | ",(0,s.jsx)(t.code,{children:"URL"})]}),"\n",(0,s.jsx)(t.h2,{id:"returns",children:"Returns"}),"\n",(0,s.jsxs)(t.p,{children:[(0,s.jsx)(t.code,{children:"Promise"}),"<",(0,s.jsx)(t.a,{href:"/docs/api/cspell-lib/interfaces/CSpellSettingsWithSourceTrace",children:(0,s.jsx)(t.code,{children:"CSpellSettingsWithSourceTrace"})}),">"]})]})}function p(e={}){const{wrapper:t}={...(0,r.R)(),...e.components};return t?(0,s.jsx)(t,{...e,children:(0,s.jsx)(d,{...e})}):d(e)}}}]);