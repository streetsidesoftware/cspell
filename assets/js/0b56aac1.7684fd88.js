"use strict";(self.webpackChunkwebsite=self.webpackChunkwebsite||[]).push([[16718],{17322:(e,s,i)=>{i.r(s),i.d(s,{assets:()=>o,contentTitle:()=>a,default:()=>p,frontMatter:()=>r,metadata:()=>l,toc:()=>c});const l=JSON.parse('{"id":"api/cspell-lib/variables/ConfigFields","title":"Variable: ConfigFields","description":"const ConfigFields: CSpellUserSettingsFields","source":"@site/docs/api/cspell-lib/variables/ConfigFields.md","sourceDirName":"api/cspell-lib/variables","slug":"/api/cspell-lib/variables/ConfigFields","permalink":"/docs/api/cspell-lib/variables/ConfigFields","draft":false,"unlisted":false,"editUrl":"https://github.com/streetsidesoftware/cspell/tree/main/website/docs/docs/api/cspell-lib/variables/ConfigFields.md","tags":[],"version":"current","frontMatter":{},"sidebar":"tutorialSidebar","previous":{"title":"Type Alias: ~~VersionLegacy~~","permalink":"/docs/api/cspell-lib/type-aliases/VersionLegacy"},"next":{"title":"Variable: ENV\\\\_CSPELL\\\\_GLOB\\\\_ROOT","permalink":"/docs/api/cspell-lib/variables/ENV_CSPELL_GLOB_ROOT"}}');var t=i(37557),n=i(46039);const r={},a="Variable: ConfigFields",o={},c=[];function d(e){const s={a:"a",blockquote:"blockquote",code:"code",h1:"h1",header:"header",p:"p",strong:"strong",...(0,n.R)(),...e.components};return(0,t.jsxs)(t.Fragment,{children:[(0,t.jsx)(s.header,{children:(0,t.jsx)(s.h1,{id:"variable-configfields",children:"Variable: ConfigFields"})}),"\n",(0,t.jsxs)(s.blockquote,{children:["\n",(0,t.jsxs)(s.p,{children:[(0,t.jsx)(s.code,{children:"const"})," ",(0,t.jsx)(s.strong,{children:"ConfigFields"}),": ",(0,t.jsx)(s.a,{href:"/docs/api/cspell-lib/type-aliases/CSpellUserSettingsFields",children:(0,t.jsx)(s.code,{children:"CSpellUserSettingsFields"})})]}),"\n"]}),"\n",(0,t.jsx)(s.p,{children:"Defined in: packages/cspell-types/dist/index.d.mts:1566"})]})}function p(e={}){const{wrapper:s}={...(0,n.R)(),...e.components};return s?(0,t.jsx)(s,{...e,children:(0,t.jsx)(d,{...e})}):d(e)}},46039:(e,s,i)=>{i.d(s,{R:()=>r,x:()=>a});var l=i(18225);const t={},n=l.createContext(t);function r(e){const s=l.useContext(n);return l.useMemo((function(){return"function"==typeof e?e(s):{...s,...e}}),[s,e])}function a(e){let s;return s=e.disableParentContext?"function"==typeof e.components?e.components(t):e.components||t:r(e.components),l.createElement(n.Provider,{value:s},e.children)}}}]);