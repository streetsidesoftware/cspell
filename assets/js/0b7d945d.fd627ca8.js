"use strict";(self.webpackChunkwebsite=self.webpackChunkwebsite||[]).push([[43752],{46039:(e,s,t)=>{t.d(s,{R:()=>o,x:()=>i});var n=t(18225);const r={},c=n.createContext(r);function o(e){const s=n.useContext(c);return n.useMemo((function(){return"function"==typeof e?e(s):{...s,...e}}),[s,e])}function i(e){let s;return s=e.disableParentContext?"function"==typeof e.components?e.components(r):e.components||r:o(e.components),n.createElement(c.Provider,{value:s},e.children)}},69483:(e,s,t)=>{t.r(s),t.d(s,{assets:()=>l,contentTitle:()=>i,default:()=>p,frontMatter:()=>o,metadata:()=>n,toc:()=>a});const n=JSON.parse('{"id":"api/cspell/functions/trace","title":"Function: trace()","description":"trace(words, options): AsyncIterableIterator\\\\","source":"@site/docs/api/cspell/functions/trace.md","sourceDirName":"api/cspell/functions","slug":"/api/cspell/functions/trace","permalink":"/docs/api/cspell/functions/trace","draft":false,"unlisted":false,"editUrl":"https://github.com/streetsidesoftware/cspell/tree/main/website/docs/docs/api/cspell/functions/trace.md","tags":[],"version":"current","frontMatter":{},"sidebar":"tutorialSidebar","previous":{"title":"Function: suggestions()","permalink":"/docs/api/cspell/functions/suggestions"},"next":{"title":"cspell","permalink":"/docs/api/cspell/globals"}}');var r=t(37557),c=t(46039);const o={},i="Function: trace()",l={},a=[{value:"Parameters",id:"parameters",level:2},{value:"words",id:"words",level:3},{value:"options",id:"options",level:3},{value:"Returns",id:"returns",level:2}];function d(e){const s={a:"a",blockquote:"blockquote",code:"code",h1:"h1",h2:"h2",h3:"h3",header:"header",p:"p",strong:"strong",...(0,c.R)(),...e.components};return(0,r.jsxs)(r.Fragment,{children:[(0,r.jsx)(s.header,{children:(0,r.jsx)(s.h1,{id:"function-trace",children:"Function: trace()"})}),"\n",(0,r.jsxs)(s.blockquote,{children:["\n",(0,r.jsxs)(s.p,{children:[(0,r.jsx)(s.strong,{children:"trace"}),"(",(0,r.jsx)(s.code,{children:"words"}),", ",(0,r.jsx)(s.code,{children:"options"}),"): ",(0,r.jsx)(s.code,{children:"AsyncIterableIterator"}),"<",(0,r.jsx)(s.code,{children:"TraceWordResult"}),">"]}),"\n"]}),"\n",(0,r.jsxs)(s.p,{children:["Defined in: ",(0,r.jsx)(s.a,{href:"https://github.com/streetsidesoftware/cspell/blob/3fc3ef2ea9def847ef20287c6238ff423e6493eb/packages/cspell/src/app/application.mts#L44",children:"cspell/src/app/application.mts:44"})]}),"\n",(0,r.jsx)(s.h2,{id:"parameters",children:"Parameters"}),"\n",(0,r.jsx)(s.h3,{id:"words",children:"words"}),"\n",(0,r.jsxs)(s.p,{children:[(0,r.jsx)(s.code,{children:"string"}),"[]"]}),"\n",(0,r.jsx)(s.h3,{id:"options",children:"options"}),"\n",(0,r.jsx)(s.p,{children:(0,r.jsx)(s.a,{href:"/docs/api/cspell/interfaces/TraceOptions",children:(0,r.jsx)(s.code,{children:"TraceOptions"})})}),"\n",(0,r.jsx)(s.h2,{id:"returns",children:"Returns"}),"\n",(0,r.jsxs)(s.p,{children:[(0,r.jsx)(s.code,{children:"AsyncIterableIterator"}),"<",(0,r.jsx)(s.code,{children:"TraceWordResult"}),">"]})]})}function p(e={}){const{wrapper:s}={...(0,c.R)(),...e.components};return s?(0,r.jsx)(s,{...e,children:(0,r.jsx)(d,{...e})}):d(e)}}}]);