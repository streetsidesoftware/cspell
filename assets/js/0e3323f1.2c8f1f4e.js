"use strict";(self.webpackChunkwebsite=self.webpackChunkwebsite||[]).push([[25350],{4002:(e,s,n)=>{n.r(s),n.d(s,{assets:()=>c,contentTitle:()=>o,default:()=>u,frontMatter:()=>r,metadata:()=>t,toc:()=>a});const t=JSON.parse('{"id":"api/cspell-types/interfaces/SuggestionsConfiguration","title":"Interface: SuggestionsConfiguration","description":"Defined in286","source":"@site/docs/api/cspell-types/interfaces/SuggestionsConfiguration.md","sourceDirName":"api/cspell-types/interfaces","slug":"/api/cspell-types/interfaces/SuggestionsConfiguration","permalink":"/docs/api/cspell-types/interfaces/SuggestionsConfiguration","draft":false,"unlisted":false,"editUrl":"https://github.com/streetsidesoftware/cspell/tree/main/website/docs/docs/api/cspell-types/interfaces/SuggestionsConfiguration.md","tags":[],"version":"current","frontMatter":{},"sidebar":"tutorialSidebar","previous":{"title":"Interface: Settings","permalink":"/docs/api/cspell-types/interfaces/Settings"},"next":{"title":"Interface: TextDocumentOffset","permalink":"/docs/api/cspell-types/interfaces/TextDocumentOffset"}}');var i=n(37557),l=n(46039);const r={},o="Interface: SuggestionsConfiguration",c={},a=[{value:"Extended by",id:"extended-by",level:2},{value:"Properties",id:"properties",level:2},{value:"numSuggestions?",id:"numsuggestions",level:3},{value:"Default",id:"default",level:4},{value:"suggestionNumChanges?",id:"suggestionnumchanges",level:3},{value:"Default",id:"default-1",level:4},{value:"suggestionsTimeout?",id:"suggestionstimeout",level:3},{value:"Default",id:"default-2",level:4}];function d(e){const s={a:"a",blockquote:"blockquote",code:"code",h1:"h1",h2:"h2",h3:"h3",h4:"h4",header:"header",hr:"hr",li:"li",p:"p",pre:"pre",strong:"strong",ul:"ul",...(0,l.R)(),...e.components};return(0,i.jsxs)(i.Fragment,{children:[(0,i.jsx)(s.header,{children:(0,i.jsx)(s.h1,{id:"interface-suggestionsconfiguration",children:"Interface: SuggestionsConfiguration"})}),"\n",(0,i.jsxs)(s.p,{children:["Defined in: ",(0,i.jsx)(s.a,{href:"https://github.com/streetsidesoftware/cspell/blob/3fc3ef2ea9def847ef20287c6238ff423e6493eb/packages/cspell-types/src/CSpellSettingsDef.ts#L286",children:"CSpellSettingsDef.ts:286"})]}),"\n",(0,i.jsx)(s.h2,{id:"extended-by",children:"Extended by"}),"\n",(0,i.jsxs)(s.ul,{children:["\n",(0,i.jsx)(s.li,{children:(0,i.jsx)(s.a,{href:"/docs/api/cspell-types/interfaces/ReportingConfiguration",children:(0,i.jsx)(s.code,{children:"ReportingConfiguration"})})}),"\n"]}),"\n",(0,i.jsx)(s.h2,{id:"properties",children:"Properties"}),"\n",(0,i.jsx)(s.h3,{id:"numsuggestions",children:"numSuggestions?"}),"\n",(0,i.jsxs)(s.blockquote,{children:["\n",(0,i.jsxs)(s.p,{children:[(0,i.jsx)(s.code,{children:"optional"})," ",(0,i.jsx)(s.strong,{children:"numSuggestions"}),": ",(0,i.jsx)(s.code,{children:"number"})]}),"\n"]}),"\n",(0,i.jsxs)(s.p,{children:["Defined in: ",(0,i.jsx)(s.a,{href:"https://github.com/streetsidesoftware/cspell/blob/3fc3ef2ea9def847ef20287c6238ff423e6493eb/packages/cspell-types/src/CSpellSettingsDef.ts#L292",children:"CSpellSettingsDef.ts:292"})]}),"\n",(0,i.jsx)(s.p,{children:"Number of suggestions to make."}),"\n",(0,i.jsx)(s.h4,{id:"default",children:"Default"}),"\n",(0,i.jsx)(s.pre,{children:(0,i.jsx)(s.code,{className:"language-ts",children:"10\n"})}),"\n",(0,i.jsx)(s.hr,{}),"\n",(0,i.jsx)(s.h3,{id:"suggestionnumchanges",children:"suggestionNumChanges?"}),"\n",(0,i.jsxs)(s.blockquote,{children:["\n",(0,i.jsxs)(s.p,{children:[(0,i.jsx)(s.code,{children:"optional"})," ",(0,i.jsx)(s.strong,{children:"suggestionNumChanges"}),": ",(0,i.jsx)(s.code,{children:"number"})]}),"\n"]}),"\n",(0,i.jsxs)(s.p,{children:["Defined in: ",(0,i.jsx)(s.a,{href:"https://github.com/streetsidesoftware/cspell/blob/3fc3ef2ea9def847ef20287c6238ff423e6493eb/packages/cspell-types/src/CSpellSettingsDef.ts#L310",children:"CSpellSettingsDef.ts:310"})]}),"\n",(0,i.jsx)(s.p,{children:"The maximum number of changes allowed on a word to be considered a suggestions."}),"\n",(0,i.jsxs)(s.p,{children:["For example, appending an ",(0,i.jsx)(s.code,{children:"s"})," onto ",(0,i.jsx)(s.code,{children:"example"})," -> ",(0,i.jsx)(s.code,{children:"examples"})," is considered 1 change."]}),"\n",(0,i.jsx)(s.p,{children:"Range: between 1 and 5."}),"\n",(0,i.jsx)(s.h4,{id:"default-1",children:"Default"}),"\n",(0,i.jsx)(s.pre,{children:(0,i.jsx)(s.code,{className:"language-ts",children:"3\n"})}),"\n",(0,i.jsx)(s.hr,{}),"\n",(0,i.jsx)(s.h3,{id:"suggestionstimeout",children:"suggestionsTimeout?"}),"\n",(0,i.jsxs)(s.blockquote,{children:["\n",(0,i.jsxs)(s.p,{children:[(0,i.jsx)(s.code,{children:"optional"})," ",(0,i.jsx)(s.strong,{children:"suggestionsTimeout"}),": ",(0,i.jsx)(s.code,{children:"number"})]}),"\n"]}),"\n",(0,i.jsxs)(s.p,{children:["Defined in: ",(0,i.jsx)(s.a,{href:"https://github.com/streetsidesoftware/cspell/blob/3fc3ef2ea9def847ef20287c6238ff423e6493eb/packages/cspell-types/src/CSpellSettingsDef.ts#L299",children:"CSpellSettingsDef.ts:299"})]}),"\n",(0,i.jsx)(s.p,{children:"The maximum amount of time in milliseconds to generate suggestions for a word."}),"\n",(0,i.jsx)(s.h4,{id:"default-2",children:"Default"}),"\n",(0,i.jsx)(s.pre,{children:(0,i.jsx)(s.code,{className:"language-ts",children:"500\n"})})]})}function u(e={}){const{wrapper:s}={...(0,l.R)(),...e.components};return s?(0,i.jsx)(s,{...e,children:(0,i.jsx)(d,{...e})}):d(e)}},46039:(e,s,n)=>{n.d(s,{R:()=>r,x:()=>o});var t=n(18225);const i={},l=t.createContext(i);function r(e){const s=t.useContext(l);return t.useMemo((function(){return"function"==typeof e?e(s):{...s,...e}}),[s,e])}function o(e){let s;return s=e.disableParentContext?"function"==typeof e.components?e.components(i):e.components||i:r(e.components),t.createElement(l.Provider,{value:s},e.children)}}}]);