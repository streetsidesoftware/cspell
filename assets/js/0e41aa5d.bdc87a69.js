"use strict";(self.webpackChunkwebsite=self.webpackChunkwebsite||[]).push([[64458],{1949:(e,r,s)=>{s.r(r),s.d(r,{assets:()=>c,contentTitle:()=>l,default:()=>d,frontMatter:()=>o,metadata:()=>t,toc:()=>a});const t=JSON.parse('{"id":"api/cspell-lib/type-aliases/ErrorLike","title":"Type Alias: ErrorLike","description":"ErrorLike string; name () => string; \\\\}","source":"@site/docs/api/cspell-lib/type-aliases/ErrorLike.md","sourceDirName":"api/cspell-lib/type-aliases","slug":"/api/cspell-lib/type-aliases/ErrorLike","permalink":"/docs/api/cspell-lib/type-aliases/ErrorLike","draft":false,"unlisted":false,"editUrl":"https://github.com/streetsidesoftware/cspell/tree/main/website/docs/docs/api/cspell-lib/type-aliases/ErrorLike.md","tags":[],"version":"current","frontMatter":{},"sidebar":"tutorialSidebar","previous":{"title":"Type Alias: ErrorEmitter()","permalink":"/docs/api/cspell-lib/type-aliases/ErrorEmitter"},"next":{"title":"Type Alias: ExclusionFunction()","permalink":"/docs/api/cspell-lib/type-aliases/ExclusionFunction"}}');var i=s(37557),n=s(46039);const o={},l="Type Alias: ErrorLike",c={},a=[];function p(e){const r={blockquote:"blockquote",code:"code",h1:"h1",header:"header",p:"p",strong:"strong",...(0,n.R)(),...e.components};return(0,i.jsxs)(i.Fragment,{children:[(0,i.jsx)(r.header,{children:(0,i.jsx)(r.h1,{id:"type-alias-errorlike",children:"Type Alias: ErrorLike"})}),"\n",(0,i.jsxs)(r.blockquote,{children:["\n",(0,i.jsxs)(r.p,{children:[(0,i.jsx)(r.strong,{children:"ErrorLike"}),": ",(0,i.jsx)(r.code,{children:"Error"})," | { ",(0,i.jsx)(r.code,{children:"message"}),": ",(0,i.jsx)(r.code,{children:"string"}),"; ",(0,i.jsx)(r.code,{children:"name"}),": ",(0,i.jsx)(r.code,{children:"string"}),"; ",(0,i.jsx)(r.code,{children:"toString"}),": () => ",(0,i.jsx)(r.code,{children:"string"}),"; }"]}),"\n"]}),"\n",(0,i.jsx)(r.p,{children:"Defined in: packages/cspell-types/dist/index.d.mts:76"})]})}function d(e={}){const{wrapper:r}={...(0,n.R)(),...e.components};return r?(0,i.jsx)(r,{...e,children:(0,i.jsx)(p,{...e})}):p(e)}},46039:(e,r,s)=>{s.d(r,{R:()=>o,x:()=>l});var t=s(18225);const i={},n=t.createContext(i);function o(e){const r=t.useContext(n);return t.useMemo((function(){return"function"==typeof e?e(r):{...r,...e}}),[r,e])}function l(e){let r;return r=e.disableParentContext?"function"==typeof e.components?e.components(i):e.components||i:o(e.components),t.createElement(n.Provider,{value:r},e.children)}}}]);