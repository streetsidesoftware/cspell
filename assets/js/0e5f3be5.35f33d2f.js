"use strict";(self.webpackChunkwebsite=self.webpackChunkwebsite||[]).push([[17178],{46039:(e,t,s)=>{s.d(t,{R:()=>r,x:()=>i});var n=s(18225);const a={},l=n.createContext(a);function r(e){const t=n.useContext(l);return n.useMemo((function(){return"function"==typeof e?e(t):{...t,...e}}),[t,e])}function i(e){let t;return t=e.disableParentContext?"function"==typeof e.components?e.components(a):e.components||a:r(e.components),n.createElement(l.Provider,{value:t},e.children)}},67744:(e,t,s)=>{s.r(t),s.d(t,{assets:()=>c,contentTitle:()=>i,default:()=>d,frontMatter:()=>r,metadata:()=>n,toc:()=>p});const n=JSON.parse('{"id":"api/cspell/type-aliases/ReplaceEntry","title":"Type Alias: ReplaceEntry","description":"ReplaceEntry: \\\\[string, string\\\\]","source":"@site/docs/api/cspell/type-aliases/ReplaceEntry.md","sourceDirName":"api/cspell/type-aliases","slug":"/api/cspell/type-aliases/ReplaceEntry","permalink":"/docs/api/cspell/type-aliases/ReplaceEntry","draft":false,"unlisted":false,"editUrl":"https://github.com/streetsidesoftware/cspell/tree/main/website/docs/docs/api/cspell/type-aliases/ReplaceEntry.md","tags":[],"version":"current","frontMatter":{},"sidebar":"tutorialSidebar","previous":{"title":"Type Alias: RegExpPatternList","permalink":"/docs/api/cspell/type-aliases/RegExpPatternList"},"next":{"title":"Type Alias: ReplaceMap","permalink":"/docs/api/cspell/type-aliases/ReplaceMap"}}');var a=s(37557),l=s(46039);const r={},i="Type Alias: ReplaceEntry",c={},p=[];function o(e){const t={blockquote:"blockquote",code:"code",h1:"h1",header:"header",p:"p",strong:"strong",...(0,l.R)(),...e.components};return(0,a.jsxs)(a.Fragment,{children:[(0,a.jsx)(t.header,{children:(0,a.jsx)(t.h1,{id:"type-alias-replaceentry",children:"Type Alias: ReplaceEntry"})}),"\n",(0,a.jsxs)(t.blockquote,{children:["\n",(0,a.jsxs)(t.p,{children:[(0,a.jsx)(t.strong,{children:"ReplaceEntry"}),": [",(0,a.jsx)(t.code,{children:"string"}),", ",(0,a.jsx)(t.code,{children:"string"}),"]"]}),"\n"]}),"\n",(0,a.jsx)(t.p,{children:"Defined in: cspell-types/dist/index.d.mts:727"})]})}function d(e={}){const{wrapper:t}={...(0,l.R)(),...e.components};return t?(0,a.jsx)(t,{...e,children:(0,a.jsx)(o,{...e})}):o(e)}}}]);