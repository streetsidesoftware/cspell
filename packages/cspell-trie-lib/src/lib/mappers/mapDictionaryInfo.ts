import { opFlatten, opMap, pipeSync } from '@cspell/cspell-pipe/sync';
import type { CharacterSetCosts, DictionaryInformation } from '../models/DictionaryInformation';
import { parseLocale } from '../models/locale';
import type { SuggestionCostMapDef } from '../models/suggestionCostsDef';
import { isDefined } from '../utils/util';
import type { EditCostsRequired } from './mapCosts';
import { mapEditCosts } from './mapCosts';
import { hunspellInformationToSuggestionCostDef } from './mapHunspellInformation';
import type { PenaltyAdjustment } from '../distance/weightedMaps';
import { calcFirstCharacterReplaceDefs, parseAccents, parseAlphabet } from './mapToSuggestionCostDef';
import type { ArrayItem } from '../types';

export function mapDictionaryInformation(dictInfo: DictionaryInformation): SuggestionCostMapDef[] {
    const _locale = dictInfo.locale;
    const locale = _locale ? parseLocale(_locale).filter((loc) => loc.isValid()) : undefined;
    const locales = locale?.map((loc) => loc.locale);
    const costs = mapEditCosts(dictInfo.costs);

    const defsEC = dictInfo.suggestionEditCosts || [];
    const defsHI = dictInfo.hunspellInformation
        ? hunspellInformationToSuggestionCostDef(dictInfo.hunspellInformation, locale)
        : [];
    return [
        ...defsEC,
        ...processAlphabet(dictInfo.alphabet, locales, costs),
        ...processAccents(dictInfo.accents, costs),
        ...defsHI,
    ];
}

function processAlphabet(
    alphabet: DictionaryInformation['alphabet'],
    locale: string[] | undefined,
    editCost: EditCostsRequired
): SuggestionCostMapDef[] {
    const csAlphabet = toCharSets(alphabet, 'a-zA-Z', editCost.baseCost);

    return [
        ...pipeSync(
            csAlphabet,
            opMap((cs) => parseAlphabet(cs, locale, editCost)),
            opFlatten()
        ),
        ...calcFirstCharacterReplaceDefs(csAlphabet, editCost),
    ];
}

function toCharSets(
    cs: string | CharacterSetCosts[] | undefined,
    defaultValue: string | undefined,
    cost: number,
    penalty?: number
): CharacterSetCosts[] {
    cs = cs ?? defaultValue;
    if (!cs) return [];

    if (typeof cs === 'string') {
        cs = [
            {
                characters: cs,
                cost,
            },
        ];
    }

    if (penalty !== undefined) {
        cs.forEach((cs) => (cs.penalty = penalty));
    }

    return cs;
}

function processAccents(
    accents: DictionaryInformation['accents'],
    editCost: EditCostsRequired
): SuggestionCostMapDef[] {
    const cs = toCharSets(accents, '\u0300-\u0341', editCost.accentCosts);
    return cs.map((cs) => parseAccents(cs, editCost)).filter(isDefined);
}

export function mapDictionaryInformationToAdjustment(dictInfo: DictionaryInformation): PenaltyAdjustment[] {
    if (!dictInfo.adjustments) return [];

    return dictInfo.adjustments.map(mapAdjustment);
}

type Adjustments = Exclude<DictionaryInformation['adjustments'], undefined>;
type Adjustment = ArrayItem<Adjustments>;

function mapAdjustment(adj: Adjustment): PenaltyAdjustment {
    const { id, regexp, penalty } = adj;
    return {
        id: id,
        regexp: new RegExp(regexp),
        penalty,
    };
}
