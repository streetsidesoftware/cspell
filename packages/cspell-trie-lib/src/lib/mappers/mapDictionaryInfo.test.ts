import { describe, expect, test } from 'vitest';

import type { DictionaryInformation } from '../models/DictionaryInformation.js';
import { mapDictionaryInformation } from './mapDictionaryInfo.js';

describe('mapDictionaryInfo', () => {
    test.each`
        info
        ${{}}
        ${di({ alphabet: 'a-e', accents: '' })}
        ${di({ alphabet: 'a-eA-E', accents: '\u0301' })}
    `('mapDictionaryInformation', ({ info }) => {
        expect(mapDictionaryInformation(info)).toMatchSnapshot();
    });
});

function di(info: DictionaryInformation): DictionaryInformation {
    return info;
}
