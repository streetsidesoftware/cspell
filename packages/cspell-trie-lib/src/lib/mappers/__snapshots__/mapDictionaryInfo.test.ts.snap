// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`mapDictionaryInfo > mapDictionaryInformation 1`] = `
[
  {
    "insDel": 100,
    "map": "ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz",
    "replace": 100,
    "swap": 100,
  },
  {
    "map": "Aa|Bb|Cc|Dd|Ee|Ff|Gg|Hh|Ii|Jj|Kk|Ll|Mm|Nn|Oo|Pp|Qq|Rr|Ss|Tt|Uu|Vv|Ww|Xx|Yy|Zz|Aa|Bb|Cc|Dd|Ee|Ff|Gg|Hh|Ii|Jj|Kk|Ll|Mm|Nn|Oo|Pp|Qq|Rr|Ss|Tt|Uu|Vv|Ww|Xx|Yy|Zz",
    "replace": 1,
  },
  {
    "map": "(^A)(^B)(^C)(^D)(^E)(^F)(^G)(^H)(^I)(^J)(^K)(^L)(^M)(^N)(^O)(^P)(^Q)(^R)(^S)(^T)(^U)(^V)(^W)(^X)(^Y)(^Z)(^a)(^b)(^c)(^d)(^e)(^f)(^g)(^h)(^i)(^j)(^k)(^l)(^m)(^n)(^o)(^p)(^q)(^r)(^s)(^t)(^u)(^v)(^w)(^x)(^y)(^z)(^)",
    "penalty": 8,
    "replace": 96,
  },
  {
    "insDel": 1,
    "map": "̴̵̶̷̸̡̢̧̨̛̖̗̘̙̜̝̞̟̠̣̤̥̦̩̪̫̬̭̮̯̰̱̲̳̹̺̻̼̀́̂̃̄̅̆̇̈̉̊̋̌̍̎̏̐̑̒̓̔̽̾̿̀́̕̚",
    "replace": 1,
  },
]
`;

exports[`mapDictionaryInfo > mapDictionaryInformation 2`] = `
[
  {
    "insDel": 100,
    "map": "ABCDEabcde",
    "replace": 100,
    "swap": 100,
  },
  {
    "map": "Aa|Bb|Cc|Dd|Ee",
    "replace": 1,
  },
  {
    "map": "(^a)(^b)(^c)(^d)(^e)(^)",
    "penalty": 8,
    "replace": 96,
  },
]
`;

exports[`mapDictionaryInfo > mapDictionaryInformation 3`] = `
[
  {
    "insDel": 100,
    "map": "ABCDEabcde",
    "replace": 100,
    "swap": 100,
  },
  {
    "map": "Aa|Bb|Cc|Dd|Ee|Aa|Bb|Cc|Dd|Ee",
    "replace": 1,
  },
  {
    "map": "(^A)(^B)(^C)(^D)(^E)(^a)(^b)(^c)(^d)(^e)(^)",
    "penalty": 8,
    "replace": 96,
  },
  {
    "insDel": 1,
    "map": "́",
    "replace": 1,
  },
]
`;
